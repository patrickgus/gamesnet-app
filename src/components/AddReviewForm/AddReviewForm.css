.AddReviewForm {
  display: flex;
  flex-direction: column;
}

.AddReviewForm label {
  color: rgb(0, 224, 224);
}

.AddReviewForm input,
textarea {
  opacity: 0.75;
}

.AddReviewForm__title {
  margin-bottom: 5px;
}

.AddReviewForm__buttons {
  display: flex;
  justify-content: space-evenly;
}

.AddReviewForm__rating {
  margin-bottom: 20px;
}

#game-rating {
  margin-left: 10px;
}
