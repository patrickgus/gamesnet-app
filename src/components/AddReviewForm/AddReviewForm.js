import React, { Component } from "react";
import { withRouter } from "react-router-dom";
import GameContext from "../../contexts/GameContext";
import GameApiService from "../../services/game-api-service";
import { Input, Required, Textarea, Button } from "../../components/Utils/Utils";
import "./AddReviewForm.css";

class AddReviewForm extends Component {
  static defaultProps = {
    match: { params: {} }
  };

  static contextType = GameContext;

  handleSumbit = e => {
    e.preventDefault();
    const { gameId } = this.props.match.params;
    const { title, rating, review } = e.target;

    GameApiService.postReview(
      gameId,
      title.value,
      Number(rating.value),
      review.value
    )
      .then(this.context.addGame)
      .then(() => {
        title.value = "";
        rating.value = "";
        review.value = "";
        this.props.history.push(`/review/${gameId}`);
      })
      .catch(this.context.setError);
  };

  handleCancel = e => {
    e.preventDefault();
    const { gameId } = this.props.match.params;
    this.props.history.push(`/review/${gameId}`);
  };

  render() {
    return (
      <form className="AddReviewForm" onSubmit={this.handleSumbit}>
        <div className="review_title">
          <label htmlFor="AddReviewForm__review_title">
            Review title <Required />
          </label>
          <Input
            name="title"
            type="text"
            required
            id="AddReviewForm__review_title"
          />
        </div>
        <div className="game_rating">
          <label htmlFor="AddReviewForm__game_rating">
            Rating <Required />
          </label>
          <select name="rating" required id="AddReviewForm__game_rating">
            <option value="1">1</option>
            <option value="2">2</option>
            <option value="3">3</option>
            <option value="4">4</option>
            <option value="5">5</option>
            <option value="6">6</option>
            <option value="7">7</option>
            <option value="8">8</option>
            <option value="9">9</option>
            <option value="10">10</option>
          </select>
        </div>
        <div className="game_review">
          <label htmlFor="AddReviewForm__game_review">
            Game review <Required />
          </label>
          <Textarea
            required
            aria-label="Write your review..."
            name="review"
            id="AddReviewForm__game_review"
            rows="15"
            placeholder="Write your review..."
          />
        </div>
        <Button type="submit">Add Review</Button>
        <Button onClick={this.handleCancel}>Cancel</Button>
      </form>
    );
  }
}

export default withRouter(AddReviewForm);
