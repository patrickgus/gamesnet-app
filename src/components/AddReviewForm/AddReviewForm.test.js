import React from "react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import AddReviewForm from "./AddReviewForm";

describe("AddReviewForm component", () => {
  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<AddReviewForm />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it("renders the form as expected", () => {
    const tree = renderer.create(<AddReviewForm />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
