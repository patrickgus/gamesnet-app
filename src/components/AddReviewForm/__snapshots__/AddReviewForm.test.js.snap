// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddReviewForm component renders the form as expected 1`] = `
<form
  className="AddReviewForm"
  onSubmit={[Function]}
>
  <div
    className="AddReviewForm__title"
  >
    <label
      htmlFor="review-title"
    >
      Review title 
      <span
        className="Required "
      >
        *
      </span>
    </label>
    <input
      className="Input "
      id="review-title"
      name="title"
      required={true}
      type="text"
    />
  </div>
  <div
    className="AddReviewForm__rating"
  >
    <label
      htmlFor="game-rating"
    >
      Rating 
      <span
        className="Required "
      >
        *
      </span>
    </label>
    <select
      id="game-rating"
      name="rating"
      required={true}
    >
      <option
        value="1"
      >
        1
      </option>
      <option
        value="2"
      >
        2
      </option>
      <option
        value="3"
      >
        3
      </option>
      <option
        value="4"
      >
        4
      </option>
      <option
        value="5"
      >
        5
      </option>
      <option
        value="6"
      >
        6
      </option>
      <option
        value="7"
      >
        7
      </option>
      <option
        value="8"
      >
        8
      </option>
      <option
        value="9"
      >
        9
      </option>
      <option
        value="10"
      >
        10
      </option>
    </select>
  </div>
  <div
    className="AddReviewForm__game-review"
  >
    <label
      htmlFor="game-review"
    >
      Game review 
      <span
        className="Required "
      >
        *
      </span>
    </label>
    <textarea
      aria-label="Write your review..."
      className="Textarea "
      id="game-review"
      name="review"
      placeholder="Write your review..."
      required={true}
      rows="15"
    />
  </div>
  <div
    className="AddReviewForm__buttons"
  >
    <button
      className="Button "
      type="submit"
    >
      Add Review
    </button>
    <button
      className="Button "
      onClick={[Function]}
    >
      Cancel
    </button>
  </div>
</form>
`;
